### DISCUSSION TITLE
How to configure nvim dap to debug a remote python program?

### DISCUSSION THREAD

**Discussion by mpas** [mfussenegger/nvim-dap#910] [Q&A] [answered]
I am trying to configure nvim dap to connect to a remote python program (running in Docker) using debugpy but seem to hit a wall on how to achieve this.

The docker container is succesfully using debugpy and I can connect remotely using VSCode and even telnet to the debug port. For this I created a `launch.json` file and tried to copy the data from that json file in my config.

See my configuration below...

```
dap.adapters.python = {
    type = 'executable';
    command = "python";
    args = { '-m', 'debugpy.adapter' };
}

dap.configurations.python = {
    {
        name = 'Python: Remote Attach',
        type = 'python',
        request = 'attach',
        connect = {
            host = 'localhost',
            port = 5678,
        },
        cwd = vim.fn.getcwd(),
        pathMappings = {
            {
                localRoot = function()
                    return vim.fn.input("Local code folder > ", vim.fn.getcwd(), "file")
                end,
                remoteRoot = function()
                    return vim.fn.input("Container code folder > ", "/", "file")
                end,
            },
        },
        justMyCode = true,
    },
}
```
With this config I seem to be unable to connect and wondering what  I am missing. Any help is highly appreciated!

**Comment by mfussenegger** [comment, marked as answer]
The adapter config needs to treat the `attach` request in a special way like this:

https://github.com/mfussenegger/nvim-dap-python/blob/65ccab83fb3d0b29ead6c765c1c52a1ed49592e8/lua/dap-python.lua#L136-L162

I'd recommend to use nvim-dap-python as it sets up the adapter correctly.

**Reply by mpas** [reply]
Thanks for the reply! Tested it with `nvim-dap-python` and this works as expected! Thanks very much!
